pkgs/data/documentation/linux-manual/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, stdenv, fetchurl, perl }:
+
+stdenv.mkDerivation rec {
+  pname = "linux-manual";
+  version = "5.15-rc3";
+
+  src = fetchurl {
+    url = "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/snapshot/linux-${version}.tar.gz";
+    sha256 = "17mqzx8v85yayriynzfs1hwjmkj6k7j13ykkcbs9d0f2b2xn4ljy";
+  };
+
+  nativeBuildInputs = [ perl ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  patchPhase = ''
+    patchShebangs --build \
+      scripts/kernel-doc \
+      scripts/split-man.pl
+  '';
+
+  installPhase = ''
+    mandir=$out/share/man/man9
+    mkdir -p $mandir
+
+    KBUILD_BUILD_TIMESTAMP=$(stat -c %Y Makefile) \
+    grep -F -l -Z \
+      --exclude-dir Documentation \
+      --exclude-dir tools \
+      -R '/**' \
+      | xargs -0 -n 256 -P $NIX_BUILD_CORES \
+        $SHELL -c '{ scripts/kernel-doc -man "$@" || :; } \
+          | scripts/split-man.pl '$mandir kernel-doc
+
+    test -f $mandir/kmalloc.9
```

Comment:
Maybe this should be part of the checkPhase?

Reply:
I considered performing this check in the `installCheckPhase`, but according to the nixpkgs manual, tests are never run for cross builds because they usually require native code execution. Since this check does not and is sensible to run even for cross builds, I think that `installPhase` is a better place for it.